p2p/net/swarm/swarm_metrics_test.go

```diff
@@ -0,0 +1,32 @@
+package swarm
+
+import (
+	"crypto/rand"
+	"testing"
+
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/network"
+
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkMetricsConnOpen(b *testing.B) {
+	b.ReportAllocs()
```

Comment:
Can we check that this is indeed some low number of allocs?

Reply:
In a unit test? Benchmarks are currently not run on CI at all.

Reply:
Yup